### PR TITLE
add kms and ppmlcontext

### PR DESCRIPTION
**Pull request by qiuxin2012**

### CODE REVIEW

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/examples/SimpleQuerySparkExample.scala [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    // Select Developer and records count
+    val developers = df.filter(df("job") === "Developer" and df("age").between(20, 40)).toDF()
+    developers.count()
+
+    Map[String, DataFrame]({
+      "developers" -> developers
+    })
+
+    // save data frame using spark kms context
+    sc.write(developers, mode = arguments.outputEncryptMode).mode("overwrite")
```
Comment (jason-dai) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/examples/SimpleQuerySparkExample.scala:63]: sc.write(developers, cyptoMode = arguments.outputEncryptMode).mode(...)

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/examples/SimpleQuerySparkExample.scala [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          EncryptIOArguments.parser.failure("miss args, please see the usage info"); null
+      }
+
+    val sc = PPMLContext.initPPMLContext("SimpleQuery", arguments.ppmlArgs())
+
+    // read kms args from spark-defaults.conf
+    // val sc = PPMLContext.initPPMLContext("SimpleQuery")
+
+    // load csv file to data frame with ppmlcontext.
+    val df = sc.read(mode = arguments.inputEncryptMode).csv(arguments.inputPath + "/people.csv")
```
Comment (jason-dai) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/examples/SimpleQuerySparkExample.scala:46]: val df = sc.read(cryptoMode = arguments.inputEncryptMode).csv(...)

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/dataframe/EncryptedDataFrameReader.scala [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ *
+ * @param sparkSession
+ * @param encryptMode
+ * @param dataKeyPlainText
+ */
+class EncryptedDataFrameReader(sparkSession: SparkSession, encryptMode: EncryptMode, dataKeyPlainText: String){
+  def csv(path: String): DataFrame = {
+    val rdd = encryptMode match {
+      case EncryptMode.PLAIN_TEXT =>
+        sparkSession.sparkContext.textFile(path)
```
Comment (jason-dai) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/dataframe/EncryptedDataFrameReader.scala:36]: why not return `sparkSession.read.cvs` directly?

**Comment by jason-dai** [comment]
Please add unit tests (e.g., comparing the behavior of encrypted read/write vs. standard read/write in Spark)

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/kms/KeyManagementService.scala [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+object KMS_CONVENTION {
+  val MODE_SIMPLE_KMS = "SimpleKeyManagementService"
+  val MODE_EHSM_KMS = "EHSMKeyManagementService"
+}
+
+trait KeyManagementService extends Supportive {
+  var _appId:String = _
+  var _appKey:String = _
+
+  def enroll():(String,String)
```
Comment (glorysdj) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/kms/KeyManagementService.scala:30]: we should remove enroll from api

**Comment by qiuxin2012** [comment]
> Please add unit tests (e.g., comparing the behavior of encrypted read/write vs. standard read/write in Spark)

The unit tests will be added in next PR.

**Pull request merged**